aws-s3/types.ml

```diff
@@ -85,9 +85,7 @@ module type Io = sig
   (**/**)
   module Net : sig
     val connect :
-      domain:Unix.socket_domain ->
-      host:string ->
-      scheme:[< `Http | `Https ] ->
+      Region.endpoint ->
```

Comment:
I suggest that `connect` is just extended with a mandatory port argument to avoid the dependency on module Region.

Reply:
Ah thanks - sorry, you mentioned this in your earlier comment.  I should be able to spend some more time on this today.

Reply:
Yes - my bad, I could not make up my mind :-). I think its easiest to see what's the best when implementing.

 I just think that the Io lib should be as independent from the rest of the lib as possible. In a distant future, we might rewrite it to use algebraic effects.

Looking forward to try it. I have been looking at the docker image for minio, and I think it relatively easy to integrate into travis ci.